packages/core/src/dialects/mariadb/data-types.ts

```diff
@@ -1,6 +1,83 @@
-import { GEOMETRY as MySqlGeometry } from '../mysql/data-types.js';
+import dayjs from 'dayjs';
+import wkx from 'wkx';
+import type { Falsy } from '../../generic/falsy.js';
+import type { GeoJson } from '../../geo-json.js';
+import { isString } from '../../utils/check.js';
+import { isValidTimeZone } from '../../utils/dayjs';
+import * as BaseTypes from '../abstract/data-types.js';
+import type { AcceptedDate, BindParamOptions } from '../abstract/data-types.js';
+
+export class FLOAT extends BaseTypes.FLOAT {
+  protected getNumberSqlTypeName(): string {
+    return 'FLOAT';
+  }
+}
+
+export class BOOLEAN extends BaseTypes.BOOLEAN {
+  toSql() {
+    return 'TINYINT(1)';
+  }
+
+  toBindableValue(value: boolean | Falsy): unknown {
+    // when binding, must be an integer
+    return value ? 1 : 0;
+  }
+}
+
+export class DATE extends BaseTypes.DATE {
+  toBindableValue(date: AcceptedDate) {
+    date = this._applyTimezone(date);
+
+    // MySQL datetime precision defaults to 0
+    const precision = this.options.precision ?? 0;
+    let format = 'YYYY-MM-DD HH:mm:ss';
+    // TODO: We should normally use `S`, `SS` or `SSS` based on the precision, but
+    //  dayjs has a bug which causes `S` and `SS` to be ignored:
+    //  https://github.com/iamkun/dayjs/issues/1734
+    if (precision > 0) {
+      format += `.SSS`;
+    }
+
+    return date.format(format);
+  }
+
+  sanitize(value: unknown, options?: { timezone?: string }): unknown {
+    if (isString(value) && options?.timezone) {
+      if (isValidTimeZone(options.timezone)) {
+        return dayjs.tz(value, options.timezone).toDate();
+      }
+
+      return new Date(`${value} ${options.timezone}`);
+    }
+
+    return super.sanitize(value);
+  }
+}
+
+export class UUID extends BaseTypes.UUID {
```

Comment:
Maybe on MariaDB >= 10.7 we can use the native UUID?

Reply:
This will add some complexity.
In https://github.com/sequelize/sequelize/pull/15407#discussion_r1056881976 you were opposed to different tests for database versions.

UUID will not be supported in the minimum version of MariaDB for quite sometime.

Reply:
Good point. Users can always extend the DataType themselves when they use MariaDB >= 10.7, although I think this adds more benefit for the users than in the other PR. I see that 10.6 is supported until July 2026 so that is still a while away. Let's not put it here then